config/health_check/health_check.go

```diff
@@ -0,0 +1,48 @@
+package health_check
```

Comment:
Is there any reason for an own package ? Otherwise a could place is still the `config` package e.g. `backend_health_check.go`.

Additionally a best practice is to avoid repetition of struct and package names.
`config.BackendHealthCheck` vs `health_check.HealthCheck`.

Further reading: https://go.dev/blog/package-names

Reply:
`health_check` was declared as a separate package to avoid import cycles which would have occurred by placing this specific file in eg. the `config` package.

The struct names have been adjusted from `HealthCheck` and `ParsedHealthCheck` to `Options` and `ParsedOptions` respectively to also better reflect their content as being configurable options for health checks when calling them with eg. `health_check.Options`.